app/controllers/documents_controller.rb

```diff
@@ -14,20 +14,19 @@ def show
   end
 
   def update
-    document = Document.find(params[:id])
-    allowed_field_names_in_contents = document.document_type_schema.fields.map(&:id)
-    document_update_params = params.require(:document).permit(:title, contents: allowed_field_names_in_contents)
-    document.update_attributes(document_update_params)
-    DocumentPublishingService.new.publish_draft(document)
-    redirect_to document, notice: "Preview creation successful"
+    @document = Document.find(params[:id])
+    @document.update_attributes(document_update_params)
+    DocumentPublishingService.new.publish_draft(@document)
+    redirect_to @document, notice: "Preview creation successful"
   rescue GdsApi::HTTPErrorResponse, SocketError => e
     Rails.logger.error(e)
-    redirect_to document, alert: "Error creating preview"
+    redirect_to @document, alert: "Error creating preview"
   end
 
 private
 
   def document_update_params
-    params.require(:document).permit(:title)
```

Comment:
Can we pass in the `document` instead of the instance variable? Instance variables imply that it's used inside a view, which isn't the case here.

Reply:
Also fine - updated.